src/Tasks/ResourceHandling/MSBuildResXReader.cs

```diff
@@ -234,10 +233,46 @@ private static void AddLinkedResource(string resxFilename, bool pathsRelativeToB
                     return;
                 }
             }
+            else if (IsByteArray(fileRefType))
+            {
+                byte[] byteArray = File.ReadAllBytes(fileName);
+
+                resources.Add(new LiveObjectResource(name, byteArray));
+                return;
+            }
+            else if (IsMemoryStream(fileRefType))
+            {
+                // See special-case handling in ResXFileRef
+                // https://github.com/dotnet/winforms/blob/689cd9c69e632997bc85bf421af221d79b12ddd4/src/System.Windows.Forms/src/System/Resources/ResXFileRef.cs#L293-L297
+                byte[] byteArray = File.ReadAllBytes(fileName);
+
+                resources.Add(new LiveObjectResource(name, new MemoryStream(byteArray)));
+                return;
+            }
 
             resources.Add(new FileStreamResource(name, fileRefType, fileName, resxFilename));
         }
 
+        /// <summary>
+        /// Does this assembly-qualified type name represent an array of bytes?
+        /// </summary>
+        /// <remarks>
+        /// We can't hard-code byte[] type name due to version number
+        /// updates & potential whitespace issues with ResX files.
+        ///
+        /// Comment and logic from https://github.com/dotnet/winforms/blob/16b192389b377c647ab3d280130781ab1a9d3385/src/System.Windows.Forms/src/System/Resources/ResXDataNode.cs#L411-L416
+        /// </remarks>
+
+        private static bool IsByteArray(string fileRefType)
+        {
+            return fileRefType.IndexOf("System.Byte[]") != -1 && fileRefType.IndexOf("mscorlib") != -1;
+        }
+
+        private static bool IsMemoryStream(string fileRefType)
+        {
+            return fileRefType.Equals(MemoryStreamTypeNameDesktopFramework, StringComparison.Ordinal) || fileRefType.Equals(MemoryStreamTypeNameCore, StringComparison.Ordinal);
```

Comment:
@ericstj Do I need to check both of these, or would just the desktop-framework flavor be sufficient? I wrote a resx using the Core `ResXResourceWriter` and it did embed

```xml
  <data name="ref" type="System.Resources.ResXFileRef, System.Windows.Forms">
    <value>whatever.txt;System.IO.MemoryStream, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e</value>
  </data>
```

Reply:
System.Private.CoreLib should not persist to source files. I think there would need to be some affordance for that when writing a resx, even if it were a resx that only needed to work with core.

Reply:
I think it should be using TypeForwardedFromAttribute to write out the original identity. That leads me to believe we do not need to support the core identity. Since nothing is writing core-specfic resx files yet in VS, I think we should wait and see how this shakes out. I would not bake System.Private.CoreLib into our code base unless we are told to do so by corefx.

Reply:
Ok, removed `System.Private.CoreLib` from here and #4638.

Reply:
Right, we had requested this ResX designer use TypeForwardedFrom.